Can't delete CSN device

I can't delete my device 'Dev1'. Attempts to do so result in 'Server Error (500)' in the browser and the following in Gunicorn's log:

```
[2019-01-22 16:38:48,187] [CSN] [ERROR] - Everynet responded with a 404 status code and the following error message: {
    "error": "Device not found"
}
[2019-01-22 16:38:48,188] [django.request] [ERROR] - Internal Server Error: /csn/device/delete
Traceback (most recent call last):
  File "/home/tfc_prod/tfc_web_venv/lib/python3.5/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/home/tfc_prod/tfc_web_venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/home/tfc_prod/tfc_web_venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/tfc_prod/tfc_web_venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/tfc_prod/tfc_web_venv/lib/python3.5/site-packages/django/contrib/auth/decorators.py", line 23, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/home/tfc_prod/tfc_web/tfc_web/csn/views.py", line 61, in delete_device
    if everynet_remove_device(lwdevice):
  File "/home/tfc_prod/tfc_web/tfc_web/csn/everynet_api.py", line 58, in everynet_remove_device
    lwdev.error_message = response.json()['message']
KeyError: 'message'
```

There is a device with that DevEUI in the Everynet console, though I have been modifying the console entries so it's possible I've broken the CSN <-> Everynet correspondence. I've successfully created and deleted a second device so I don't know what's going on here.

--- a/tfc_web/csn/forms.py
+++ b/tfc_web/csn/forms.py
@@ -60,6 +60,7 @@ class LWDeviceForm(forms.Form):
         lw_application = cleaned_data.get("lw_application")
         if 'user_id' not in lw_application.info or self.user.id != lw_application.info['user_id']:
             raise ValidationError("You are not authorise to add this device to the application selected")
+        cleaned_data['dev_eui'] = cleaned_data['dev_eui'].lower()
         cleaned_data['sensor_id'] = cleaned_data['dev_eui']
         cleaned_data['sensor_type'] = 'lorawan'
         cleaned_data['user_id'] = self.user.id
@@ -130,7 +131,19 @@ class LWDeviceFormExtended(LWDeviceForm):
                               help_text="The application key is used to derive the two session keys during the "
                                         "activation procedure. The key is 128 bits in length represented by a "
                                         "32-character hexadecimal string.")
-
+    
+    def clean(self):
+        cleaned_data = super(LWDeviceFormExtended, self).clean()
+        if 'dev_addr' in cleaned_data:
+            cleaned_data['dev_addr'] = cleaned_data['dev_addr'].lower()
+        if 'nwkskey' in cleaned_data:
+            cleaned_data['nwkskey'] = cleaned_data['nwkskey'].lower()
+        if 'appskey' in cleaned_data:
+            cleaned_data['appskey'] = cleaned_data['appskey'].lower()
+        if 'app_key' in cleaned_data:
+            cleaned_data['app_key'] = cleaned_data['app_key'].lower()
+        return cleaned_data
+    
     def _post_clean(self):
         if 'activation_type' in self.cleaned_data and self.cleaned_data['activation_type'] == "otaa":
             self.cleaned_data.pop('dev_addr', None)

